parlai/scripts/train_model.py

```diff
@@ -242,6 +241,45 @@ def load_eval_worlds(agent, opt, datatype):
     return worlds
 
 
+def _run_single_eval(opt, valid_world, max_exs):
+    # run evaluation on a single world
+    valid_world.reset()
+
+    cnt = 0
+    max_cnt = max_exs if max_exs > 0 else float('inf')
+    while not valid_world.epoch_done() and cnt < max_cnt:
+        valid_world.parley()
+        if cnt == 0 and opt['display_examples']:
+            print(valid_world.display() + '\n~~')
+            print(valid_world.report())
+        cnt += valid_world.opt['batchsize']
+
+    valid_report = valid_world.report()
+    valid_world.reset()  # make sure world doesn't remember valid data
+
+    return valid_report
+
+
+def _aggregate_reports(reports, valid_worlds):
```

Comment:
are these substantially different? Why not just import it from the train_model. (or vice versa)? Should aggregate (where there's already an aggregator) go into metrics?

Reply:
can't put it in metrics because these are being aggregated post-report. can put a utility function in metrics and import it here though. might be cleaner.